modules/onboarding/src/main/kotlin/edu/stanford/spezi/module/onboarding/consent/SignaturePad.kt

```diff
@@ -62,13 +60,13 @@ internal fun SignaturePad(
             }
         )
 
-        if (uiState.firstName.value.isNotBlank() && uiState.lastName.value.isNotBlank()) {
+        if ((uiState.name.givenName ?: "").isNotBlank() && (uiState.name.familyName ?: "").isNotBlank()) {
             Spacer(modifier = Modifier.height(Spacings.medium))
             Text("Signature:")
             SignatureCanvas(
                 paths = uiState.paths.toMutableList(),
-                firstName = uiState.firstName.value,
-                lastName = uiState.lastName.value,
+                firstName = uiState.name.givenName ?: "",
```

Comment:
see above - recomposition would work though, since the onAction updates the state with a new PersonNameComponents instance every time, since PersonNameComponents only contains constants.